### ISSUE TITLE
idea behind this algorihtm

### ISSUE THREAD

**Issue by DonnyLee** [DonnyLee/HorseAlgorithm#1]
Hello. Since i completely forgot work on this... i will leave an issue for future me.

This algorithm was introduced to one of my professor while i was grinding for master's degree back in the day. He found my idea funny but i was serious about it. There was a competition to implement and tweak existing movement algorithm to find shortest way from point A to B in the void efficiently. We took the BAT-Algorithm to approach, we utterly programmed the behaviour of a bat with algorithm suggested by MR Yang et.el.(https://www.inderscienceonline.com/doi/abs/10.1504/IJBIC.2013.055093).
We simulated ultra sonic emit and read ability of a bat, to find the target (food). It was huge success and a lot of math and fun. So i decided to program an horse algorithm, at least a concept in pseudo code.

It is going to be one of simple mobility model to simulate general behaviour of horses. Why horses? i saw behaviour of horses in real life and documentary and found it interesting that... 
1. they seem to find herd back when scattered badly, 
2. in time of peace herd stay in close proximity, following the alpha male or female horse.
3. They find water to drink.
4. Random grazing behaviour
just https://horse-canada.com/magazine/miscellaneous/10-amazing-facts-equine-smell/#:~:text=The%20horse's%20large%2C%20flexible%20nostrils,predators%20or%20other%20environmental%20changes.

TL; DR;
Horse algorithm, simulate horse and horses as heard, maybe as an mobility model, visualize in browser maybe.

we won the competition btw.

**Comment by DonnyLee** [comment]
I reverse engineered former fitness lib used in MasterProjekt similar swarm Bat algorithm.
https://github.com/DonnyLee/MasterProjekt_BA/tree/master/src/main/java/fitness
